data_structures/linked_list/singly_link_list_deletion.c

```diff
@@ -18,38 +19,75 @@ struct node *createnode()  // function to create node
     return (t);
 }
 ////////////////////////////////////////////////////////
-void insert()  // function to insert at first location
+int insert(int pos)  // function to insert at first location
 {
-    struct node *p;
-    p = createnode();
-    printf("\nenter the number to insert");
-    scanf("%d", &p->info);
-    p->link = NULL;
-    if (start == NULL)
+    struct node *new;
+    new = createnode();
+    printf("\nenter the number to be inserted :");
+    scanf("%d", &new->info);
+    if(pos==1)
     {
-        start = p;
-    }
-    else
-    {
-        p->link = start;
-        start = p;
+    new->link = NULL;
+       if (start == NULL)
+      {
+        start = new;
+      }
+      else
+      {
+        new->link = start;
+        start = new;
+       }
     }
+    else{
+       struct node *pre=start;
+       for(int i=1;i<pos-1;i++)
+       {
+           if (pre==NULL){
+             printf("Position not found!");
+             return;
+            }
+           pre=pre->link;
+       }
+          new->link=pre->link;
+          pre->link=new;
+          
+       }
+    return 0;
 }
 ///////////////////////////////////////////////////////////
-void deletion()  // function to delete from first position
+int deletion(int pos)  // function to delete from any position
```

Comment:
Same here.

Reply:
Similarly here in line 81(when input location is not present to delete) to return to function, I need to use "int"